src/domain/Processors/VehicleProcessor.ts

```diff
@@ -82,7 +83,12 @@ export abstract class VehicleProcessor<T extends Vehicle> {
         primaryVrm,
         enums.SEARCHCRITERIA.VRM
       );
-      if (primaryVrmRecords.length > 0) {
+      
+      const allTechRecordArrays = primaryVrmRecords.map(record => record.techRecord);
+      const allTechRecords = new Array() as Array<ITechRecord>;
+      allTechRecordArrays.map(record => record.forEach(x => allTechRecords.push(x)));
```

Comment:
```suggestion
      const allTechRecords = primaryVrmRecords.map(wrapper => wrapper.techRecord).flat();
```
If we are allowed to use ES2019, then use `.flatMap()`.

Reply:
I wanted to use flatmap, but I'll wait for the go ahead to upgrade to ES2019 before doing so

Reply:
My bad, `flat()` also requires ES2019.
Thus, use this please:
```suggestion
      const allTechRecords = primaryVrmRecords
        .map(wrapper => wrapper.techRecord)
        .reduce((accumulator, value) => accumulator.concat(value), []);
```